gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -382,12 +365,31 @@ public synchronized void setActive(boolean active) {
               ServiceMetricNames.FAILED_FLOW_METER));
         }
 
+        GobblinServiceQuotaManager quotaManager = new GobblinServiceQuotaManager(config);
+        // Before initializing the DagManagerThreads check which dags are currently running before shutdown
+        Set<String> runningDags = new ConcurrentHashSet<>();
+        for (Dag<JobExecutionPlan> dag: dagStateStore.getDags()) {
+          for (DagNode<JobExecutionPlan> dagNode: dag.getNodes()) {
+            if (DagManagerUtils.getExecutionStatus(dagNode) == RUNNING) {
+              runningDags.add(DagManagerUtils.generateDagId(dagNode));
+              // Add all the currently running Dags to the quota limit per user
+              try {
+                quotaManager.checkQuota(dagNode);
+              } catch (IOException e) {
+                // Quota is somehow exceeded with currently running jobs, we should never hit this state normally
+                // but we should avoid stalling the entire service
+                log.error(String.format("Quota exceeded during initialization in DagManager for job name: %s",
```

Comment:
this is exceeding quota of entire dag manager not per user quota right?

Reply:
It would be a per user quota, bit since this is happening at startup we should technically never exceed the quota since it would be loading the previously running jobs, which can't be exceeding the quota.